packages/trpc/server/routers/viewer/organizations/update.handler.ts

```diff
@@ -14,47 +19,99 @@ type UpdateOptions = {
   input: TUpdateInputSchema;
 };
 
-export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
-  const { logo, bio, orgId, password } = input;
-
-  const userMembership = await prisma.membership.findFirst({
-    where: {
-      userId: ctx.user.id,
-      teamId: orgId,
-    },
-    select: {
-      userId: true,
-      role: true,
-    },
-  });
-
-  // TODO test this check works
-  if (!userMembership || userMembership.role !== MembershipRole.OWNER)
-    throw new TRPCError({ code: "BAD_REQUEST", message: "not_authorized" });
-
+// @Leo - looking at this the more i think this should be its own route. This is for updating a org that is used elsehwere and
+// resetting password kinda polutes updateing an entire org. It could lead to API parameter manipulation down the line.
+async function updateAdminPassword(userId: number, password?: string) {
   if (password) {
     const hashedPassword = await hashPassword(password);
     await prisma.user.update({
       where: {
-        id: ctx.user.id,
+        id: userId,
       },
       data: {
         password: hashedPassword,
       },
     });
+  }
+}
```

Comment:
Out of scope of this PR - handled from merge.

I think should be abstracted outside of the update handler @leog

Reply:
merge onboarding branch, I moved it out to its own router